test/loading.jl

```diff
@@ -14,6 +14,12 @@ include_string_test_func = include_string("include_string_test() = @__FILE__", t
 
 @test isdir(@__DIR__)
 @test @__DIR__() == dirname(@__FILE__)
+let exename = `$(Base.julia_cmd()) --precompiled=yes --startup-file=no`
+    wd = string('"', pwd(), '"')
+    @test readchomp(`$exename -E "@__DIR__" -i`) == wd
```

Comment:
use `println(@__DIR__)`, so that it doesn't get handed to `show` to escape all of the characters (or set `wd` to `sprint(show, pwd())`)